tests/test_client.py

```diff
@@ -0,0 +1,237 @@
+#     Valerio Cosentino <valcos@bitergia.com>
+#
+
+import os
+import sys
+import time
+import unittest
+
+import httpretty
+import pkg_resources
+import requests
+
+# Hack to make sure that tests import the right packages
+# due to setuptools behaviour
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), ".."))
+pkg_resources.declare_namespace('perceval.backends')
+
+from perceval.client import HttpClient, RateLimitHandler
+
+
+CLIENT_API_URL = "https://gateway.marvel.com/v1/"
+CLIENT_SPIDERMAN_URL = "https://gateway.marvel.com/v1/public/characters/1"
+CLIENT_SUPERMAN_URL = "https://gateway.marvel.com/v1/public/characters/2"
+CLIENT_BATMAN_URL = "https://gateway.marvel.com/v1/public/characters/3"
+CLIENT_IRONMAN_URL = "https://gateway.marvel.com/v1/public/characters/4"
+
+
+class MockedClient(HttpClient, RateLimitHandler):
+
+    def __init__(self, base_url, default_sleep_time=0.1, max_retries=1):
+
+        super().__init__(base_url, default_sleep_time=default_sleep_time, max_retries=max_retries)
+        super().setup_rate_limit_handler()
+
+
+class TestHttpClient(unittest.TestCase):
+    """Http client tests """
+
+    def test_initialization(self):
+        """Test whether attributes are initializated"""
+
+        client = MockedClient(CLIENT_API_URL)
+
+        self.assertEqual(client.base_url, CLIENT_API_URL)
+        self.assertEqual(client.max_retries, 1)
+        self.assertEqual(client.max_retries_on_connect, HttpClient.MAX_RETRIES_ON_CONNECT)
+        self.assertEqual(client.max_retries_on_read, HttpClient.MAX_RETRIES_ON_READ)
+        self.assertEqual(client.max_retries_on_redirect, HttpClient.MAX_RETRIES_ON_REDIRECT)
+        self.assertEqual(client.max_retries_on_read, HttpClient.MAX_RETRIES_ON_READ)
+        self.assertEqual(client.max_retries_on_status, HttpClient.MAX_RETRIES_ON_STATUS)
+        self.assertEqual(client.status_forcelist, HttpClient.DEFAULT_STATUS_FORCE_LIST)
+        self.assertEqual(client.method_whitelist, HttpClient.DEFAULT_METHOD_WHITELIST)
+        self.assertEqual(client.raise_on_redirect, HttpClient.DEFAULT_RAISE_ON_REDIRECT)
+        self.assertEqual(client.raise_on_status, HttpClient.DEFAULT_RAISE_ON_STATUS)
+        self.assertEqual(client.respect_retry_after_header, HttpClient.DEFAULT_RESPECT_RETRY_AFTER_HEADER)
+        self.assertEqual(client.default_sleep_time, 0.1)
+
+        self.assertIsNotNone(client.session)
+
+    @httpretty.activate
+    def test_close_session(self):
+        """Test wheter the session is properly closed"""
+
+        output = "success"
+        httpretty.register_uri(httpretty.GET,
+                               CLIENT_SPIDERMAN_URL,
+                               body=output,
+                               status=200)
+
+        client = MockedClient(CLIENT_API_URL)
+        response = client.fetch(CLIENT_SPIDERMAN_URL)
+        self.assertEqual(response.headers['connection'], 'close')
+
+    @httpretty.activate
+    def test_fetch(self):
```

Comment:
Slipt these checks into three independent tests. Therefore, a failure in one of the tests doesn't stop for testing the others.